### PR TITLE
Raise an error if any field has duplicates in the yaml

### PR DESCRIPTION
**Pull request by nickbp**
- Specified 'memory' or 'count' twice? That's an error
- Also an explicit check disallowing both 'volume' and 'volumes' at once

### CODE REVIEW

File: sdk/scheduler/src/test/resources/invalid-pod-name.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ pods:
     resource-sets:
       meta-data-resources:
         cpus: 0.1
-        mem: 512
+        memory: 512
```
Comment (mrbrowning) [inline sdk/scheduler/src/test/resources/invalid-pod-name.yml]: Should this file be called `invalid-resource-name.yml`?

File: sdk/scheduler/src/test/resources/invalid-task-name.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ pods:
     resource-sets:
       meta-data-resources:
         cpus: 0.1
-        mem: 512
+        memory: 512
```
Comment (mrbrowning) [inline sdk/scheduler/src/test/resources/invalid-task-name.yml]: Similar issue with YAML file name.

File: sdk/scheduler/src/test/resources/invalid-pod-name.yml (hunk 1 above)
Comment (nickbp) [inline sdk/scheduler/src/test/resources/invalid-pod-name.yml]: (These misspelled fields were being missed in the tests -- they were throwing a parse exception but not the one the tests were meant for)

Comment (nickbp) [inline sdk/scheduler/src/test/resources/invalid-pod-name.yml]: I think it's accurate -- the duplicate 'meta-data' pod name is the issue

File: sdk/scheduler/src/test/resources/invalid-task-name.yml (hunk 2 above)
Comment (nickbp) [inline sdk/scheduler/src/test/resources/invalid-task-name.yml]: In this case it's two tasks both named 'meta-data-task'

File: sdk/scheduler/src/test/resources/invalid-pod-name.yml (hunk 1 above)
Comment (gabrielhartmann) [inline sdk/scheduler/src/test/resources/invalid-pod-name.yml]: Are the changes you made to the tests sufficient to make them fail now if we put in unknown properties?

Comment (nickbp) [inline sdk/scheduler/src/test/resources/invalid-pod-name.yml]: No the tests were just written to check for ANY JsonMappingException:

**Pull request merged**